### DISCUSSION TITLE
[Messenger] Minimum privileges for database auto-setup

### DISCUSSION THREAD

**Discussion by gnito-org** [symfony/symfony#44970] [Q&A]
When using ``auto_setup`` for the database transport in Messenger, is the minimum data definition privileges  ``CREATE``, ``ALTER``, and ``INDEX``? Or is ``DROP`` also required?

**Comment by ghost** [comment]
Why don't you check this on your own? Run worker and see if auto setup works. If not then add the privileges one by one if required.

**Reply by gnito-org** [reply]
Thank you for your helpful advice. Unfortunately that will not work, because that will only test if CREATE is required, which is already an established fact. What is unknown is whether auto-setup will also attempt to perform ALTER of DROP actions later on, if the Messenger Doctrine transport gets modified in future releases.

**Reply by ghost** [reply]
> Unfortunately that will not work, because that will only test if CREATE is required

For doctrine transport auto setup is executed only on TableNotFoundException so ALTER and DROP will never happen as long as the table exists in the database which it does.

> if the Messenger Doctrine transport gets modified in future releases.

If it is modified and it is a BC break then it will not be in the minor version. For a major version you always have to investigate the UPGRADE.md file.
